spinnaker/data_source_spinnaker_canary_config.go

```diff
@@ -45,6 +46,6 @@ func datasourceCanaryConfig() *schema.Resource {
 				},
 			},
 		},
-		Read: resourceSpinnakerCanaryConfigRead,
+		ReadContext: resourceSpinnakerCanaryConfigRead,
```

Comment:
[note]
[More Support for context.Context](https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html#more-support-for-context-context)

>  for example, the helper/schema.CreateFunc, helper/schema.UpdateFunc, helper/schema.ReadFunc, and helper/schema.DeleteFunc types all got context-aware versions (helper/schema.CreateContextFunc, helper/schema.UpdateContextFunc, helper/schema.ReadContextFunc, helper/schema.DeleteContextFunc, respectively). The old versions are now deprecated, and we recommend you upgrade to the context-aware versions to get their improvements, ...